apps/mobile/src/screens/GalleryScreen/GalleryEditorNftSelector.tsx

```diff
@@ -0,0 +1,80 @@
+import { RouteProp, useNavigation, useRoute } from '@react-navigation/native';
+import { Suspense, useCallback } from 'react';
+import { View } from 'react-native';
+
+import { NftSelectorHeader } from '~/components/NftSelector/NftSelectorHeader';
+import { NftSelectorToolbar } from '~/components/NftSelector/NftSelectorToolbar';
+import { NftSelectorWrapper } from '~/components/NftSelector/NftSelectorWrapper';
+import { useNftSelector } from '~/components/NftSelector/useNftSelector';
+import { RootStackNavigatorParamList, RootStackNavigatorProp } from '~/navigation/types';
+import { NftSelectorLoadingSkeleton } from '~/screens/NftSelectorScreen/NftSelectorLoadingSkeleton';
+import { NftSelectorPickerGrid } from '~/screens/NftSelectorScreen/NftSelectorPickerGrid';
+
+export function GalleryEditorNftSelector() {
+  const {
+    searchQuery,
+    setSearchQuery,
+    ownershipTypeFilter,
+    setFilter,
+    networkFilter,
+    setNetworkFilter,
+    sortView,
+    setSortView,
+    sync,
+    isSyncing,
+    isSyncingCreatedTokens,
+  } = useNftSelector();
+  const navigation = useNavigation<RootStackNavigatorProp>();
+
+  const route = useRoute<RouteProp<RootStackNavigatorParamList, 'NftSelectorGalleryEditor'>>();
+
+  const handleSelectNft = useCallback(
+    (tokenId: string) => {
+      navigation.navigate({
+        name: 'GalleryEditor',
+        params: {
+          galleryId: route.params.galleryId,
+          stagedTokens: [tokenId],
+        },
+        merge: true,
+      });
+    },
+    [navigation, route.params.galleryId]
+  );
+
+  return (
+    <NftSelectorWrapper ownershipTypeFilter={ownershipTypeFilter} isFullscreen>
+      <View className="gap-8">
+        <NftSelectorHeader title="Select item to add" />
+        <NftSelectorToolbar
+          searchQuery={searchQuery}
+          setSearchQuery={setSearchQuery}
+          ownershipTypeFilter={ownershipTypeFilter}
+          setFilter={setFilter}
+          networkFilter={networkFilter}
+          setNetworkFilter={setNetworkFilter}
+          sortView={sortView}
+          setSortView={setSortView}
+          isSyncing={isSyncing}
+          isSyncingCreatedTokens={isSyncingCreatedTokens}
+          handleSync={sync}
+        />
+      </View>
+      <View className="flex-grow flex-1 w-full">
+        <Suspense fallback={<NftSelectorLoadingSkeleton />}>
+          <NftSelectorPickerGrid
+            searchCriteria={{
+              searchQuery,
+              ownerFilter: ownershipTypeFilter,
+              networkFilter: networkFilter,
+              sortView,
+            }}
+            onRefresh={sync}
+            onSelect={handleSelectNft}
+            onSelectNftGroup={() => {}}
```

Comment:
1. let's make sure to memoize `searchCriteria`
2. let's pass in `noop` into L74